gas-bound-caller/contracts/GasBoundCaller.sol

```diff
@@ -83,8 +85,18 @@ contract GasBoundCaller {
         }
 
         assembly {
-            // We just relay the return data from the call.
-            return(add(returnData, 0x20), mload(returnData))
+            // This place does interfere with the memory layout, however, it is done right before
+            // the `return` statement, so it is safe to do.
+            // We need to transform `bytes memory returnData` into (bytes memory returndata, gasSpentOnPubdata)
+            // `bytes memory returnData` is encoded as `length` + `data`.
+            // We need to prepend it with 0x40 and `pubdataGas`.
+            //
+            // It is assumed that the position of returndata is >= 0x40, since 0x40 is the free memory pointer location.
+            mstore(sub(returnData, 0x40), 0x40)
+            mstore(sub(returnData, 0x20), pubdataGas)
+            let returndataLen := add(mload(returnData), 0x60)
+
+            return(sub(returnData, 0x40), returndataLen)
```

Comment:
Nit/low:

`returndataLen` must always be divisible by 32. That's a part of ABI specification. But `abi.decode` can parse data even though its length isn't divisible by 32. 

I am not sure about the implications of it, but If the caller propagates data instead of parsing it, the staticcall can return something that is not parsable by ethers.